backend/src/polydash/polygon/top_peers.py

```diff
@@ -0,0 +1,60 @@
+from fastapi import APIRouter
+from pony.orm import desc, db_session
+
+from polydash.miners_ratings.model import MinerRisk
+from polydash.polygon.deanon.model import DeanonNodeByBlock, DeanonNodeByTx, PeerToIP
+
+TOP_NODES_LIST_SIZE = 10
+
+top_peers_router = router = APIRouter(
+    prefix="/top-peers",
+    tags=["W3Router"],
+    responses={404: {"description": "Not found"}},
+)
+
+
+@router.get("/")
+@db_session
+async def get_top_peers() -> list[str]:
+    memorized_peer_ids = set()
+    # Iterating over a list is fine every time, since it typically is very small...
+    new_top_nodes = []
+    # build a new list, including the IP addresses
+    for node in MinerRisk.select().order_by(desc(MinerRisk.risk)):
+        # try to get possible peer IDs of the node using two tables
+        deanoned_nodes = []
+        for table in (DeanonNodeByBlock, DeanonNodeByTx):
+            deanoned_nodes.extend(
+                table.select(signer_key=node.pubkey)
+                .order_by(desc(table.confidence)))
+        if not deanoned_nodes:
+            # we don't know peer ID of this node, moving on
+            continue
+
+        # try to get the peer ID of this node which we don't know yet
+        final_deanoned_node = None
+        for deanoned_node in deanoned_nodes:
+            if deanoned_node.peer_id not in memorized_peer_ids:
+                final_deanoned_node = deanoned_node
+                memorized_peer_ids.add(final_deanoned_node.peer_id)
+                break
+        if final_deanoned_node is None:
+            continue
+
+        # now, try to get IP address of that node
+        if not (node_ip := PeerToIP.select(peer_id=final_deanoned_node.peer_id)
+                .order_by(desc(PeerToIP.id))
+                .first()):
+            # we don't know IP of this node, moving on
+            continue
+
+        # we have found IP of that node, memorize it if it wasn't in the list before
+        if node_ip.ip in new_top_nodes:
+            continue
+
+        new_top_nodes.append(node_ip.ip)
```

Comment:
In the previous implementation we were adding the RPC port instead of the original one in that IP address. Don't we want to do that anymore?

Reply:
Nah, we don't need it now. The thing will instead send it to Bor to relay it over p2p